docs/docs/dev_docs/contracts/syntax/storage.md

```diff
@@ -1,40 +1,533 @@
+
+### `insert_from_public`
+
+While we don't support private functions to directly alter public storage, the opposite direction is possible. The `insert_from_public` allow public function to insert notes into private storage. This is very useful when we want to support private function calls that must have been initiated in public, such as shielding or the like.
+
+#include_code insert_from_public /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+The usage is rather straight-forward and very similar to using the `insert` method with the difference that this one is called in public functions.
+
+#include_code insert_from_public /yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+### `assert_contains_and_remove`
+
+This function is used to check existence of a note and then remove it without having read the note ahead of time. This can be useful for cases where the user would is providing all the information needed, such as cases where the note was never emitted to the network and thereby available to the wallet.
+
+#include_code assert_contains_and_remove /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+<!---
+@LHerskind
+I don't see why this one is actually needed and could not be deprecated down the line. 
+Allow insertions into the rpc database and this one can be removed.
+-->
+
+### `assert_contains_and_remove_publicly_created`
+
+Like above, this is used to ensure that the message exists in the data tree and then consume it. However, it differs slightly since there is currently a difference between notes that have been inserted from public and private execution. This means that you currently must use this function to consume and nullify a note that was created in a public function. This will be fixed in the future. 
+
+#include_code assert_contains_and_remove_publicly_created /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+While this might look intimidating, the use of the function is rather easy, and is used in the following way:
+
+#include_code assert_contains_and_remove_publicly_created /yarn-project/noir-contracts/src/contracts/token_contract/src/main.nr rust
+
+The reason we are not reading this note ahead of time is that no [encrypted log](./events.md#encrypted-events) was emitted for this note, since it was created in public thereby making the encrypted log useless (everyone saw the content ahead of time).
+
+### `remove`
+
+Will remove a note from the set if it previously have been read from storage. That being that you have fetched it through a `get_notes` call. This is useful when you want to remove a note that you have previously read from storage to not have to read it again. If you recall from earlier, we are emitting a nullifier when reading values to make sure that they are up to date. 
+
+#include_code remove /yarn-project/aztec-nr/aztec/src/state_vars/set.nr rust
+
+An example of how to use this operation is visible in the `easy_private_state`:
```

Comment:
Do we have an explanation or example anywhere in the docs about why we have easy private state or how to use it? Might be good to link to it, or create a resource if we don't.

Reply:
Not sure there is a separate example, but generally it was to make it "easier" to work to use than sets directly such that they from dev perspective behaves kinda like a singular value. Should probably be refactored as well, will create a separate issue for this.